### ISSUE TITLE
quarto, rmd no longer registered in nvim-treesitter

### ISSUE THREAD

**Issue by PMassicotte** [jmbuhr/otter.nvim#179]
https://github.com/nvim-treesitter/nvim-treesitter/commit/da61d31a3d51f38a78a739392aabf79e7b2f523f

**Comment by jmbuhr** [comment]
Oh, interesting. I assume this means that associations of this kind should live in their respective plugins, but we can check back with @clason on the commit about the intended direction of the change. I can't do so right now, because commenting on commits is not loading for me on mobile. Would you do that?

I can add the fix to quarto-nvim. Did you do so for Rmd via R.nvim?

Along the same vein we should look towards creating a treesitter grammar for Rmd and Qmd to accommodate the differences from plain markdown. There is a way to do inheritance for grammars, but I didn't have the time to look into it more deeply.

**Comment by clason** [comment]
This is intentional since this breaks the automated commentstring detection in Neovim.

**Comment by jmbuhr** [comment]
Ok, thanks! Will the commentstrings also break if we add the association via `vim.treesitter.language.register` or is this indeed the intended way?

**Comment by clason** [comment]
I believe it would also break, at least on Neovim master.

Rmd/quarto is _not_ Markdown, even though it looks very similar to the untrained eye. You really should write a dedicated parser for it. (It's possible to extend other parsers so you don't have to start from scratch.)

**Comment by PMassicotte** [comment]
> I can add the fix to quarto-nvim. Did you do so for Rmd via R.nvim?

I have initiated a PR here: https://github.com/R-nvim/R.nvim/pull/239

It seems to be working fine with the little testing I have done so far.

**Comment by PMassicotte** [comment]
That is also interesting that I am using qmd and md a lot in my day-to-day work and never had this commentstrings issue.

**Comment by jmbuhr** [comment]
Agreed, I started with a Qmd parser at some point, just as a fork of the markdown one, but didn't have the time to finish it properly.

**Comment by jmbuhr** [comment]
Relying on the markdown grammar was always just a temporary solution that happend to hold for a surprising amount of time, so I was anticipating this.

**Comment by PMassicotte** [comment]
I agree. We have patched this in R.nvim, but we should properly extend markdown tree-sitter.

**Comment by clason** [comment]
If the difference is only the comment string (or at least only at the top level), you could just replace that and still keep the `markdown_inline` injections. That would vastly simplify things.

**Comment by jmbuhr** [comment]
In the short term this will be my first test once I'm back from vacation, though in the long run we'll want to extend both the block and inline parser to accommodate differences (e.g. surrounding `~` denote subscript and `~~` denote strikethrough in pandoc's markdown, while the commonmark grammar recognizes both as strikethrough).

**Comment by clason** [comment]
Well, there was an "if" in this statement ;)

**Comment by PMassicotte** [comment]
@jmbuhr Let me know, I would be interested contributing.

**Comment by jmbuhr** [comment]
@PMassicotte Amazing! I'll add you once I have a repo.

**Comment by PMassicotte** [comment]
Awesome! Thank you and enjoy your vacation.

**Comment by jmbuhr** [comment]
I created one at https://github.com/jmbuhr/tree-sitter-pandoc-markdown. It is built as an extension of the markdown grammar (though currently doesn't do any actual extending), but still requires some files from the original markdown grammar due to the custom scanner and their built setup (see readme). Might get simpler in the future if extending TS grammars becomes more common place.

I'll add you as a collaborator. @PMassicotte

**Comment by PMassicotte** [comment]
Just back to the office, I will try that this morning.
